homeassistant/components/skybell/camera.py

```diff
@@ -1,106 +1,69 @@
 """Camera support for the Skybell HD Doorbell."""
 from __future__ import annotations
 
-from datetime import timedelta
-import logging
-
-import requests
 import voluptuous as vol
 
-from homeassistant.components.camera import PLATFORM_SCHEMA, Camera
+from homeassistant.components.camera import (
+    PLATFORM_SCHEMA,
+    Camera,
+    CameraEntityDescription,
+)
+from homeassistant.config_entries import ConfigEntry
 from homeassistant.const import CONF_MONITORED_CONDITIONS
 from homeassistant.core import HomeAssistant
 import homeassistant.helpers.config_validation as cv
+from homeassistant.helpers.entity import EntityDescription
 from homeassistant.helpers.entity_platform import AddEntitiesCallback
-from homeassistant.helpers.typing import ConfigType, DiscoveryInfoType
-
-from . import DOMAIN as SKYBELL_DOMAIN, SkybellDevice
-
-_LOGGER = logging.getLogger(__name__)
 
-SCAN_INTERVAL = timedelta(seconds=90)
-
-IMAGE_AVATAR = "avatar"
-IMAGE_ACTIVITY = "activity"
-
-CONF_ACTIVITY_NAME = "activity_name"
-CONF_AVATAR_NAME = "avatar_name"
+from .const import DOMAIN, IMAGE_ACTIVITY, IMAGE_AVATAR
+from .coordinator import SkybellDataUpdateCoordinator
+from .entity import SkybellEntity
 
+# Deprecated in Home Assistant 2022.6
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend(
     {
         vol.Optional(CONF_MONITORED_CONDITIONS, default=[IMAGE_AVATAR]): vol.All(
             cv.ensure_list, [vol.In([IMAGE_AVATAR, IMAGE_ACTIVITY])]
         ),
-        vol.Optional(CONF_ACTIVITY_NAME): cv.string,
-        vol.Optional(CONF_AVATAR_NAME): cv.string,
+        vol.Optional("activity_name"): cv.string,
+        vol.Optional("avatar_name"): cv.string,
```

Comment:
Please leave the constants in until the deprecation.  Let's not make this PR any bigger.